packages/cadl-ranch-specs/http/resiliency/srv-driven/main.tsp

```diff
@@ -0,0 +1,196 @@
+
+  There are three concepts that should be clarified:
+  1. Client spec version: refers to the spec that the client is generated from. 'v1' is a client generated from old.tsp and 'v2' is a client generated from main.tsp.
+  2. Service deployment version: refers to a deployment version of the service. 'v1' represents the initial deployment of the service with a single api version. 'v2' represents the new deployment of a service with multiple api versions
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  With the above two calls, we test the following configurations from this service spec:
+  - A client generated from the second service spec can call the second deployment of a service with api version v1
+  - A client generated from the second service spec can call the second deployment of a service with api version v2 with the updated changes
+
+  Tests that we can grow up an operation from accepting one required parameter to accepting a required parameter and an optional parameter.
+  """)
+  @route("/from-one-required")
+  @doc("Operation that grew up from accepting one required parameter to accepting a required parameter and an optional parameter.")
+  @get
+  fromOneRequired(
+    @doc("I am a required parameter")
+    @query
+    parameter: string,
+
+    @added(Versions.v2)
+    @doc("I'm a new input optional parameter")
+    @query
+    `new-parameter`?: string
+  ): NoContentResponse;
+
+  @scenario
+  @scenarioDoc("""
+  Need the following two calls:
+  - Pass in `serviceDeploymentVersion="v2"` and `apiVersion="v1"` with query parameter `parameter="optional"`.
+  - Pass in `serviceDeploymentVersion="v2"` and `apiVersion="v2"` with query parameter `parameter="optional"` and query parameter `new-parameter="new"`.
+
+  There are three concepts that should be clarified:
+  1. Client spec version: refers to the spec that the client is generated from. 'v1' is a client generated from old.tsp and 'v2' is a client generated from main.tsp.
+  2. Service deployment version: refers to a deployment version of the service. 'v1' represents the initial deployment of the service with a single api version. 'v2' represents the new deployment of a service with multiple api versions
+  3. Api version: The initial deployment of the service only supports api version 'v1'. The new deployment of the service supports api versions 'v1' and 'v2'.
+
+  With the above two calls, we test the following configurations from this service spec:
+  - A client generated from the second service spec can call the second deployment of a service with api version v1
+  - A client generated from the second service spec can call the second deployment of a service with api version v2 with the updated changes
+
+  Tests that we can grow up an operation from accepting one optional parameter to accepting two optional parameters.
+  """)
+  @route("/from-one-optional")
```

Comment:
Do we need another test case of "add a new optional parameter to the operation with one required and one optional parameter"?

Reply:
we removed that one because didn't see this being a different use case, do you have an example were this could be treated differently?